### PR TITLE
doc: update documentation for new release

### LINKED ISSUE #984: Update supported releases, capability levels docs

### PR DESCRIPTION
**Pull request by jsilvela**
Closes #984 

Signed-off-by: Jaime Silvela <jaime.silvela@enterprisedb.com>

### CODE REVIEW

File: docs/src/supported_releases.md [hunk 1]
```diff
@@ -79,11 +80,10 @@ by the CloudNativePG maintainers).
...
 
-| Version         | Release Date       | End of Life              | Supported Kubernetes Versions |
-|-----------------|--------------------|--------------------------|-------------------------------|
-| 1.18.0          | October 25, 2022   | April 13, 2023           | 1.23, 1.24, 1.25              |
-| 1.19.0          | January 12, 2023   | -                        |                               |
-| 1.20.0          | March 13, 2023     | -                        |                               |
+| Version         | Release Date       | Code Freeze      | End of Life              | Supported Kubernetes Versions |
+|-----------------|--------------------|------------------|--------------------------|-------------------------------|
+| 1.19.0          | January 19, 2023   | January 12, 2023 | -                        |                               |
+| 1.20.0          | March 13, 2023     | -                | -                        |                               |
```
Comment (mnencia) [inline docs/src/supported_releases.md:86]:
```suggestion
| 1.20.0          | March 13, 2023     | -                | -                        |                               |
| 1.21.0          | May 15, 2023       | -                | -                        |  
```

File: docs/src/supported_releases.md [hunk 2]
```diff
@@ -65,6 +65,7 @@ Git tags for versions are prepended with `v`.
 
 | Version         | Currently Supported  | Release Date      | End of Life              | Supported Kubernetes Versions | Tested, but not supported |
 |-----------------|----------------------|-------------------|--------------------------|-------------------------------|---------------------------|
+| 1.18.x          | Yes                  | November 10, 2022 | ~ April 13, 2023         | 1.23, 1.24, 1.25              |                           |
```
Comment (mnencia) [inline docs/src/supported_releases.md:68]:
```suggestion
| 1.18.x          | Yes                  | November 10, 2022 | ~ April 13, 2023         | 1.23, 1.24, 1.25              | 1.22                      |
```

File: docs/src/index.md [hunk 3]
```diff
@@ -31,7 +31,8 @@ The [source code repository is in Github](https://github.com/cloudnative-pg/clou
 
 ## Supported Kubernetes distributions
 
-CloudNativePG requires Kubernetes 1.22 or higher. For more information, please
+CloudNativePG requires Kubernetes 1.23 or higher for the latest 1.18 release.
+Previous releases can also support Kubernetes 1.22. For more information, please
```
Comment (mnencia) [inline docs/src/index.md:35]: I do not like this change. It is not clear and not generic enough. I would revert this change and restore the previous version.

Comment (mnencia) [inline docs/src/index.md:35]:
```suggestion
CloudNativePG requires Kubernetes 1.22 or higher. For more information, please
```

**Review by mnencia — APPROVED** [review]

**Review by gbartolini — APPROVED** [review]
